### PR TITLE
Wip kill warnings

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@xiexingguo true. will drop it from my pr.

**Comment by tchaikov** [comment]
@gregsfortytwo could you help review it?

File: src/test/librbd/fsx.cc [hunk 1]
```diff
@@ -1405,12 +1405,12 @@ dowrite(unsigned offset, unsigned size)
 		       (debug &&
 		       (monitorstart == -1 ||
 			(offset + size > monitorstart &&
-			(monitorend == -1 || offset <= monitorend))))))
+			 (monitorend == -1 || (long)offset <= monitorend))))))
```
Comment (gregsfortytwo) [inline src/test/librbd/fsx.cc]: You should use C++-style conversion operators here: static_cast<long>(offset).

Comment (gregsfortytwo) [inline src/test/librbd/fsx.cc]: Oh, unless fsx.cc already has this casting style — it's not our code.

**Comment by gregsfortytwo** [comment]
Other than the casting style on the last two commits this looks good to me, but you should really get somebody who's more comfortable with autotools to check over those bits.

(Also there's a failing test but I suspect that's just the build bot flaking out.)

File: src/test/librbd/fsx.cc (hunk 1 above)
Comment (tchaikov) [inline src/test/librbd/fsx.cc]: yeah, it already has this C style casting. and i found that `fsx.cc` originally comes from AFS and was written in C. but we adapted to our usage and changed it to fsx.**cc**.

**Comment by tchaikov** [comment]
@gregsfortytwo thanks for your review.

i fixed the `(char *)` cast in the `unittest_ipaddr` change.

yeah, test (test/test_objectstore_memstore.sh) also fails in master. see http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-tarball-trusty-amd64-basic/#origin/master . guess sage will take care of it.

**Comment by tchaikov** [comment]
since all of the commits are LGTM'ed. i am taking the liberty to label this pr with "needs-qa".

**Comment by tchaikov** [comment]
see http://qa-proxy.ceph.com/teuthology/sage-2016-01-14_19:17:26-rados-wip-sage-testing2---basic-smithi/29637/teuthology.log

again
```
2016-01-17T08:39:37.300 INFO:tasks.workunit.client.0.smithi053.stderr:+ ceph_test_rados_api_io
...
2016-01-17T08:39:38.407 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.SimpleWrite
2016-01-17T08:39:38.407 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:38.407 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:38.408 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:38.408 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.754 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.SimpleWrite (1346 ms)
2016-01-17T08:39:39.754 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.TooBig
2016-01-17T08:39:39.754 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.754 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.755 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.755 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.767 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.TooBig (13 ms)
2016-01-17T08:39:39.767 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.ReadTimeout
2016-01-17T08:39:39.767 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.768 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.768 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.768 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.780 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.ReadTimeout (14 ms)
2016-01-17T08:39:39.781 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.RoundTrip
2016-01-17T08:39:39.781 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.781 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.782 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.782 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.795 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.RoundTrip (15 ms)
2016-01-17T08:39:39.796 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.OverlappingWriteRoundTrip
2016-01-17T08:39:39.796 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.796 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.797 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.797 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.810 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.OverlappingWriteRoundTrip (15 ms)
2016-01-17T08:39:39.810 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.WriteFullRoundTrip
2016-01-17T08:39:39.811 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.811 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.811 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.812 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.825 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.WriteFullRoundTrip (14 ms)
2016-01-17T08:39:39.825 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.AppendRoundTrip
2016-01-17T08:39:39.825 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.825 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.826 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.826 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.838 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.AppendRoundTrip (14 ms)
2016-01-17T08:39:39.839 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.TruncTest
2016-01-17T08:39:39.839 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.839 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.839 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.840 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.853 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.TruncTest (14 ms)
2016-01-17T08:39:39.853 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.RemoveTest
2016-01-17T08:39:39.853 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.853 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.854 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.854 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.866 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.RemoveTest (14 ms)
2016-01-17T08:39:39.867 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.XattrsRoundTrip
2016-01-17T08:39:39.867 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.867 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.868 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.868 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.880 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.XattrsRoundTrip (14 ms)
2016-01-17T08:39:39.880 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.RmXattr
2016-01-17T08:39:39.881 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.881 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.881 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.882 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.893 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.RmXattr (14 ms)
2016-01-17T08:39:39.894 INFO:tasks.workunit.client.0.smithi053.stdout:[ RUN      ] LibRadosIo.XattrIter
2016-01-17T08:39:39.894 INFO:tasks.workunit.client.0.smithi053.stdout:test/librados/TestCase.cc:265: Failure
2016-01-17T08:39:39.894 INFO:tasks.workunit.client.0.smithi053.stdout:Value of: requires
2016-01-17T08:39:39.895 INFO:tasks.workunit.client.0.smithi053.stdout:  Actual: true
2016-01-17T08:39:39.895 INFO:tasks.workunit.client.0.smithi053.stdout:Expected: false
2016-01-17T08:39:39.907 INFO:tasks.workunit.client.0.smithi053.stdout:[  FAILED  ] LibRadosIo.XattrIter (13 ms)
2016-01-17T08:39:41.136 INFO:tasks.workunit.client.0.smithi053.stdout:[----------] 12 tests from LibRadosIo (1500 ms total)
```

**Comment by tchaikov** [comment]
@liewegas, thanks for testing this pr, and sorry for not testing the change for the tests. it failed in the test in a batch you helped to run against it. and it failed again!

i fixed it in 03ead74a0dcf12c81a276410453b65c6aec2d500 and will run the failed test in sepia and post the test result here.

@xiexingguo could you help review 03ead74 ?

**Comment by tchaikov** [comment]
@liewegas test passed at http://pulpito.ceph.com/kchai-2016-01-17_22:29:42-rados:thrash-wip-kill-warnings---basic-multi/32492/.

**Comment by tchaikov** [comment]
updated the typos in commit comment.

**Pull request merged**